GeeksCoreLibrary/Modules/HealthChecks/Services/DatabaseHealthService.cs

```diff
@@ -0,0 +1,82 @@
+    private readonly HttpContext httpContext;
+    private readonly HealthChecksSettings healthChecksSettings;
+    
+    public DatabaseHealthService(IDatabaseConnection databaseConnection, IHttpContextAccessor httpContextAccessor, IOptions<HealthChecksSettings> healthChecksSettings)
+    {
+        this.databaseConnection = databaseConnection;
+        httpContext = httpContextAccessor.HttpContext;
+        this.healthChecksSettings = healthChecksSettings.Value;
+    }
+    
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
+        CancellationToken cancellationToken = new CancellationToken())
+    {
+        databaseConnection.ClearParameters();
+       
+        // A new query to check the max active connections from the DatabaseHealthCheck
+        var query =
+            $"SELECT COUNT(*) AS active_connections FROM information_schema.PROCESSLIST;";
+        var datatable = await databaseConnection.GetAsync(query);
+
+        if (datatable.Rows.Count == 0)
+        {
+            return await Task.FromResult(HealthCheckResult.Unhealthy("No data found"));
+        }
+         
+        var healthCheckConnections = healthChecksSettings.MaximumDatabaseConnections;
+        var healthCheckConnectionsTime = healthChecksSettings.MaximumConnectionsInTime;
+
+        // If no value is set, we are skipping this test.
+        if (healthCheckConnections > 0)
+        {
+            // Retrieve the count of active connections.
+            var activeConnections = Convert.ToInt32(datatable.Rows[0]["active_connections"]); 
+            
+            // Check if the number of active connections exceeds the limit.
+            if (activeConnections > healthCheckConnections) 
+            {
+                return await Task.FromResult(HealthCheckResult.Unhealthy(activeConnections +" databases connected, Too many database connections"));
+            }
+        }
+        
+        // If no value is set, we are skipping this test.
+        if (healthCheckConnectionsTime > 0)
+        {    
+            // Query to check the max open connections in time from the DatabaseHealthCheck.
+            query =  $"SELECT ID, USER, HOST, TIME AS connection_time_in_sec FROM information_schema.PROCESSLIST";
+            datatable = await databaseConnection.GetAsync(query);
+
+            if (datatable.Rows.Count == 0)
+            {
+                return await Task.FromResult(HealthCheckResult.Unhealthy("No data found"));
+            }
+            
+            // Retrieve the count of open connections in time.
+            var connectionTime = Convert.ToInt32(datatable.Rows[0]["connection_time_in_sec"]);
+
+            // Check if the time from open connections exceeds the limit.
+            if (connectionTime > healthCheckConnectionsTime)
+            {
+                return await Task.FromResult(HealthCheckResult.Unhealthy("To many seconds of connection, time is too long")); 
```

Comment:
Please also remove the `Task.FromResult` here, because it's not needed. The message is also a little bit broken English (no offense). Here's my suggestion:
```c#
return HealthCheckResult.Unhealthy($"There is at least one opened connection that has been open for more than {healthCheckConnectionsTim} seconds. There most likely is an issue with this connection.");
```